### PR TITLE
Switch existings tests to use equalsIgnoringHashCodes and add more tests.

### PR DESCRIPTION
**Pull request by jacob314**
I'm open to keeping some of these tests on a local branch if we are concerned they will cause
too many busy work changes as widgets are updated. Just let me know which tests to keep out of the repo.
These tests were useful in refactoring the toStringDeep code but are perhaps a bit excessive
outside of that context.

### CODE REVIEW

File: packages/flutter/test/rendering/paragraph_test.dart [hunk 1]
```diff
@@ -177,5 +176,29 @@ void main() {
...
+      paragraph.toStringDeep(),
+      equalsIgnoringHashCodes(
+        'RenderParagraph#00000 NEEDS-LAYOUT NEEDS-PAINT DETACHED\n'
+        ' │ parentData: null\n'
+        ' │ constraints: null\n'
+        ' │ size: MISSING\n'
+        ' ╘═╦══ text ═══\n'
+        '   ║ TextSpan:\n'
+        '   ║   "I polished up that handle so carefullee\n'
+        'That now I am the Ruler of the Queen\'s Navee!"\n'
```
Comment (Hixie) [inline packages/flutter/test/rendering/paragraph_test.dart]: can you add a TODO(ianh) to fix this?

**Comment by Hixie** [comment]
LGTM. I'm fine with checking this in.

I think we would be better served by specific tests of each class, though.

File: packages/flutter/test/widgets/animated_container_test.dart [hunk 2]
```diff
@@ -65,6 +65,41 @@ void main() {
...
+        ' │ creator: DecoratedBox ← Container ←\n'
+        ' │   AnimatedContainer-[GlobalKey#00000] ← [root]\n'
+        ' │ parentData: <none>\n'
+        ' │ constraints: BoxConstraints(w=800.0, h=600.0)\n'
+        ' │ size: Size(800.0, 600.0)\n'
+        ' │ decoration:\n'
+        ' │   color: Color(0xff0000ff)\n'
+        ' │ configuration: ImageConfiguration(bundle:\n'
+        ' │   PlatformAssetBundle#00000(), devicePixelRatio: 1.0, platform:\n'
+        ' │   macos)\n'
```
Comment (Hixie) [inline packages/flutter/test/widgets/animated_container_test.dart]: aren't you going to have the same issue here?

**Comment by Hixie** [comment]
You can just force the platform to be whatever you want it to be, I think. Doesn't it come from the Theme?

**Comment by jacob314** [comment]
Fixed the issue that was causing the platform to not match the platform specified for tests in a separate CL.
Updated all tests to now expect android and verified the tests pass on macos.

File: packages/flutter/test/widgets/overlay_test.dart [hunk 3]
```diff
@@ -25,5 +25,53 @@ void main() {
       ],
     ));
     expect(didBuild, isTrue);
+    final RenderObject theater = overlayKey.currentContext.findRenderObject();
+
+    // toStringDeep output is missing a trailing line break.
```
Comment (Hixie) [inline packages/flutter/test/widgets/overlay_test.dart]: todo?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/widgets/overlay_test.dart (hunk 3 above)
Comment (jacob314) [inline packages/flutter/test/widgets/overlay_test.dart]: done

**Pull request merged**